server/server.go

```diff
@@ -240,7 +240,12 @@ func newServer(ctx context.Context, c Config, rotationStrategy rotationStrategy)
 	handleWithCORS("/keys", s.handlePublicKeys)
 	handleFunc("/auth", s.handleAuthorization)
 	handleFunc("/auth/{connector}", s.handleConnectorLogin)
-	handleFunc("/callback", s.handleConnectorCallback)
+	r.HandleFunc(path.Join(issuerURL.Path, "/callback"), func(w http.ResponseWriter, r *http.Request) {
+		// Strip the X-Remote-User header to prevent security issues on
+		// misconfigured authproxy connector setups.
+		r.Header.Del("X-Remote-User")
```

Comment:
Let's future proof this against other headers:

```go
for key := range r.Header {
    if strings.HasPrefix(strings.ToLower(key), "x-remote-") {
        r.Header.Del(key)
    }
}
```

Reply:
Done.